INDI/INDI/indiserver.c

```diff
@@ -2315,8 +2317,13 @@ static int readFromDriver(DvrInfo *dp)
     return (shutany ? -1 : 0);
 }
 
```

Comment:
The semantic changes start here.

```main``` becomes ```__INDISERVER_MAIN__```, which will always be a macro.

If that macro is not #defined, then it will revert back to ```main``` and INDI server will build normally i.e. as a C-only application.

The code under ```/apps/xindiserver/callable_indiserver.c``` #includes this ```indiserver.c``` after @define-ing ```__INDISERVER_MAIN__``` as ```callable_indiserver```, which ```callable_indiserver``` is then called from within ```xindiserver.hpp```; see below.